Service/src/main/java/Repository/ThreadModelRep/ThreadRepositoryImpl.java

```diff
@@ -0,0 +1,592 @@
+                UserImpl user =   RepositoryFactory.getRepository()
+                        .getUserById(rs.getInt("trainer_id"));
+                result.put(groupId,
+                        new Group()
+                                .withId(groupId)
+                                .withName(rs.getString("name"))
+                                .withTrainer(user)
+                                .withStudents(
+                                        studentsFromGroup(groupId)
+                                )
+                                .withTheam(theamFromGroup(groupId))
+                        );
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    @Override
+    public HashMap<Integer, Theams> allTheams() {
+        HashMap<Integer, Theams> result = new HashMap<>();
+        try (Connection connection = datasourse.getConnection()) {
+            PreparedStatement ps = connection.prepareStatement("select * from theam");
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                int theamId = rs.getInt("id");
+                result.put(theamId,
+                        new Theams()
+                                .withId(theamId)
+                                .withValue(rs.getString("theam_name"))
+                );
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    @Override
+    public HashMap<Trainer, List<Salary>> trainerSalary() {
+        HashMap<Trainer, List<Salary>> result = new HashMap<>();
+        ArrayList<Integer> trainersIDList = new ArrayList<>(RepositoryFactory.getRepository().allTrainer().keySet());
+        try (Connection connection = datasourse.getConnection()) {
+            for (Integer trainerId:
+                    trainersIDList) {
+                UserImpl user = RepositoryFactory.getRepository().getUserById(trainerId);
+                Trainer trainer = (Trainer) new Trainer()
+                        .withSalary(new ArrayList<>())
+                        .withId(user.getId())
+                        .withLogin(user.getLogin())
+                        .withName(user.getName())
+                        .withPassword(user.getPassword())
+                        .withRole(user.getRole());
+
+                PreparedStatement ps = connection.prepareStatement(
+                        "select * from salary where trainer_id = "+trainerId);
+                ResultSet rs = ps.executeQuery();
+                while (rs.next()) {
+                        int salaryValue = rs.getInt("salary_value");
+                         if (!rs.wasNull())
+                           trainer
+                            .addSalary(
+                                    new Salary()
+                                            .withValue(salaryValue)
+                            );
```

Comment:
В Java более уместно такое форматирование
```suggestion
                           trainer.addSalary(new Salary()
                                            .withValue(salaryValue));
```

Reply:
Исправил.